### ISSUE TITLE
FollowMouse and swaylock: wrong window active

### ISSUE THREAD

**Issue by jech** [labwc/labwc#1943]
A very minor issue, but one that I find annoying.

I've got the following in my `rc.xml`:
```
  <focus>
    <followMouse>yes</followMouse>
    <followMouseRequiresMovement>no</followMouseRequiresMovement>
  </focus>
```
I also have the following in my `autostart` script:
```
swayidle -w \
	timeout 550 'swaylock -f -c 000000' \
	timeout 600 'wlopm --off \*' \
	resume 'wlopm --on \*' \
	before-sleep 'swaylock -f -c 000000' >/dev/null 2>&1 &
```
This works fairly well, in the sense that the active window is always the one under the mouse, except in one case: if swayidle triggers and swaylock locks the system, it is sometimes the case that the active window is not the one under the mouse.  Moving the mouse clears the discrepancy, but I occasionally end up typing in the wrong window, which is mildly annoying.

**Comment by tokyo4j** [comment]
I guess your issue is partially fixed with 83fa919, though that change was not intended fix your case.

Prior to 83fa919, the topmost window was always focused. Then 83fa919 changes it to re-focus the window under cursor when `followMouse` is yes and `followMouseRequiresMovement` is no. This change is a side-effect of adding the call to `desktop_focus_view()` in order to update cursor shape on unlock.

However, I think the behavior which focuses the topmost window on unlock may be surprising also for users not setting `followMouse` to yes and `followMouseRequiresMovement` to no. Perhaps we should remember the focused surface on unlock then restore it on unlock.

**Comment by Consolatis** [comment]
> Perhaps we should remember the focused surface on lock then restore it on unlock.

That sounds sensible to me. However we need to ensure that the surface still exists when unlocking, otherwise we throw a dangling pointer around.

**Comment by jech** [comment]
> I guess your issue is partially fixed with 83fa919

Possible, since I'm running 0.7.2.
